recon/src/libp2p/tests.rs

```diff
@@ -243,7 +250,120 @@ async fn responder_model_error() {
+        BTreeStoreErrors::default(),
+        BTreeStoreErrors::default(),
+        bob_model_store,
+        BTreeStoreErrors::default(),
+    );
+
+    let fut = async move {
+        swarm1.listen().with_memory_addr_external().await;
+        swarm2.connect(&mut swarm1).await;
+
+        // Expect interests to sync twice in a row since models fail to sync
+        let (
+            [p1_e1, p1_e2, p1_e3, p1_e4, p1_e5, p1_e6, p1_e7, p1_e8, p1_e9, p1_e10, p1_e11, p1_e12],
+            [p2_e1, p2_e2, p2_e3, p2_e4, p2_e5, p2_e6, p2_e7, p2_e8, p2_e9, p2_e10, p2_e11, p2_e12],
+        ): ([crate::libp2p::Event; 12], [crate::libp2p::Event; 12]) =
+            libp2p_swarm_test::drive(&mut swarm1, &mut swarm2).await;
+
+        let events = [
+            [
+                &p1_e1, &p1_e2, &p1_e3, &p1_e4, &p1_e5, &p1_e6, &p1_e7, &p1_e8, &p1_e9, &p1_e10,
+                &p1_e11, &p1_e12,
+            ],
+            [
+                &p2_e1, &p2_e2, &p2_e3, &p2_e4, &p2_e5, &p2_e6, &p2_e7, &p2_e8, &p2_e9, &p2_e10,
+                &p2_e11, &p2_e12,
+            ],
+        ];
+
+        for ev in events.iter().flatten() {
+            info!("{:?}", ev);
+        }
+        let stream_sets: Vec<_> = events
+            .iter()
+            .map(|peer_events| {
+                peer_events
+                    .iter()
+                    .map(|ev| {
+                        let crate::libp2p::Event::PeerEvent(PeerEvent { status, .. }) = ev;
+                        match status {
+                            PeerStatus::Waiting | PeerStatus::Stopped => None,
+                            PeerStatus::Synchronized { stream_set }
+                            | PeerStatus::Started { stream_set }
+                            | PeerStatus::Failed { stream_set } => Some(*stream_set),
+                        }
+                    })
+                    .collect::<Vec<Option<crate::libp2p::StreamSet>>>()
+            })
+            .collect();
+        let expected_stream_set_order = vec![
+            // First interests sync
+            Some(StreamSet::Interest),
+            Some(StreamSet::Interest),
+            // First model sync
+            Some(StreamSet::Model),
+            Some(StreamSet::Model),
+            // Second interests sync
+            Some(StreamSet::Interest),
+            Some(StreamSet::Interest),
+            // Second model sync with initial short backoff
+            Some(StreamSet::Model),
+            Some(StreamSet::Model),
+            // Third interests sync
+            Some(StreamSet::Interest),
+            Some(StreamSet::Interest),
+            // Third model sync is skipped because the backoff pushed it past the interests sync
+            Some(StreamSet::Interest),
+            Some(StreamSet::Interest),
+        ];
+        assert_eq!(
+            stream_sets,
+            vec![expected_stream_set_order.clone(), expected_stream_set_order]
+        );
+        assert_eq!(
+            p2_e4,
```

Comment:
`p2_e8` would also be a failure, right? Don't have to check for it, just confirming my understanding of how the test works.